keps/sig-api-machinery/0015-dry-run.md

```diff
@@ -150,3 +151,27 @@ A dry-run request should behave as close as possible to a regular
 request. Attempting to dry-run create an existing object will result in an
 `AlreadyExists` error to be returned. Similarly, if a dry-run update is
 performed on a non-existing object, a `NotFound` error will be returned.
+
+## kubectl
+
+For the `kubectl` client integration with server-side dry-run, we will pass
+the `dryRun` query parameter by reading the user's intent from a flag.
+
+For beta, we use `--server-dry-run` for `kubectl apply` to exercise
+server-side apply. This flag will be deprecated next release, then removed in 2 releases.
+
+For GA, we'll use the existing `--dry-run` flag available on `kubectl apply`.j
+
+Currently, the `--dry-run` flag is a boolean for subcommands including
+`kubectl apply` and `kubectl create`.
+
+We'll extend the flag to accept strings for new options
+for selecting client-side and server-side behavior: `client`, `server`, and
+`none`.
+
+For backwards compatibility, we'll continue to default the value for `--dry-run`
```

Comment:
I would explicitely call out in this KEP the code that you'll use to check compatibility, something along the lines of ...
```golang
func ParseDryRun(dryRunFlag string) (DryRunEnum, error) {
  b, err := strconv.ParseBool(dryRunFlag)
  if err != nil { // The flag is not a boolean
    switch dryRunFlag {
    case "client":
      return DryRunClient, nil
    case "server":
      return DryRunServer, nil
    default:
      return DryRunNo, fmt.Errorf(`Invalid dry-run value (%v), must be boolean, or "client", or "server"`, dryRunFlag)
    }
  }
  if b { // flag was a boolean, and indicates true, run client-side
    return DryRunClient, nil
  }
  return DryRunNo, nil
}
```

Reply:
Maybe not necessarily code, but the general flow - yeah, that'd make sense, although not required.

Reply:
Done.